.github/workflows/pr-scan.yml

```diff
@@ -0,0 +1,87 @@
+name: PR Security Scan
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  workflow_dispatch:
+jobs:
+  scan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        submodules: 'recursive'
+    - name: Build the Docker image
+      uses: docker/build-push-action@v5
+      with: 
+        context: .
+        push: false
+        load: true
+        tags: pr_image:${{ github.sha }}
+    - name: Run Trivy vulnerability scanner on image
+      uses: aquasecurity/trivy-action@master
+      with:
+        image-ref: 'pr_image:${{ github.sha }}'
+        format: 'json'
+        output: 'trivy-image-results.json'
+        severity: 'CRITICAL,HIGH'
+    - name: Run Trivy vulnerability scanner on source code
+      uses: aquasecurity/trivy-action@master
+      with:
+        scan-type: 'fs'
+        scan-ref: '.'
+        format: 'json'
+        output: 'trivy-fs-results.json'
+        severity: 'CRITICAL,HIGH'
+    - name: Process and post Trivy scan results
+      uses: actions/github-script@v7
```

Comment:
How come we aren't using `github/codeql-action/upload-sarif@v2` as mentioned in https://github.com/aquasecurity/trivy-action?tab=readme-ov-file#using-trivy-with-github-code-scanning

Reply:
For a PR-triggered workflow like this, the primary goal of this job is to call out any new potential vulnerabilities introduced by the changes in the pull request. `github/codeql-action/upload-sarif@v2` would be used in a nightly run to keep the security tab updated.